### ISSUE TITLE
Unpacking-non-sequence raises error for Callable in collections.abc

### ISSUE THREAD

**Issue by TylerYep** [PyCQA/pylint#4895]
### Bug description
This instance of the bug is a regression in pylint 2.10.2. It does not appear in pylint 2.9.

```python
from dataclasses import dataclass
from collections.abc import Callable


@dataclass
class Metric:
    function: Callable[..., tuple[int, int]]

    def update(self):
        _, _ = self.function()
```

Related issues:
https://github.com/PyCQA/pylint/issues/4137
https://github.com/PyCQA/pylint/issues/4696
https://github.com/PyCQA/pylint/issues/4803

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:9:8: E0633: Attempting to unpack a non-sequence defined at line 502 of _collections_abc (unpacking-non-sequence)

------------------------------------------------------------------
Your code has been rated at 1.67/10 (previous run: 1.67/10, +0.00)
```

### Expected behavior

Shouldn't throw this error.

### Pylint version

```shell
pylint 2.10.2
astroid 2.7.2
Python 3.9.2
```

### OS / Environment

Mac OS

**Comment by Pierre-Sassoulas** [comment]
Thank for opening the issue, on ubuntu I'm getting:
```
************* Module a
a.py:7:14: E1136: Value 'Callable' is unsubscriptable (unsubscriptable-object)
a.py:7:28: E1136: Value 'tuple' is unsubscriptable (unsubscriptable-object)

------------------------------------
Your code has been rated at -6.67/10
```
If it's MacOs specific, we might need to do #4759 first.

**Comment by cdce8p** [comment]
@Pierre-Sassoulas Try it with at least Python 3.9 or add `from __future__ import annotations`.
